src/gui/mzroll/mainwindow.cpp

```diff
@@ -2030,6 +2025,17 @@ void MainWindow::_warnIfNISTPolarityMismatch()
     }
 }
 
+void MainWindow::setLastLoadedDatabase(QString filename)
+{
+    QFileInfo fileInfo(filename);
+    bool isMSPFile = filename.endsWith("msp", Qt::CaseInsensitive);
+    bool isSPTXTFile = filename.endsWith("sptxt", Qt::CaseInsensitive);
+    bool nistFile = isMSPFile || isSPTXTFile;
+    bool smallerThan2Mb = fileInfo.size() < 2000000;
+    if (!nistFile || smallerThan2Mb)
```

Comment:
@saifulbkhan Why are we only saving the last set library if it's not a spectral library?

Reply:
The check is meant to make sure that if the file _is_ a spectral library, it has to be less than 2 Mb. The assumption here is that non-spectral databases are usually within the 2 Mb limit. Let me know if you think that should be changed.

Reply:
@saifulbkhan My bad. I thought it was an 'and' condition